src/main.cpp

```diff
@@ -2080,6 +2081,47 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+Consensus::VersionBitsConditionChecker checkCSV(Consensus::DEPLOYMENT_CSV);
+Consensus::VersionBitsConditionChecker* const softforkcheck[] = {&checkCSV};
+
+int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Params& params)
+{
+    LOCK(cs_main);
+    int32_t nVersion = 0;
+
+    for (unsigned int i = 0; i < sizeof(softforkcheck) / sizeof(softforkcheck[0]); i++) {
+        if (softforkcheck[i]->IsSetFor(pindexPrev, params)) {
+            nVersion |= softforkcheck[i]->Mask(params);
+        }
+    }
+
+    if (nVersion == 0) {
```

Comment:
I mean, if we are going to wait to deploy the first SF to merge it, I think it makes even more sense that we never return Consensus::VERSIONBITS_LAST_OLD_BLOCK_VERSION here.

>> That would mean that versionbits on itself is a change, and lack of
adoption of versionbits by miners would then cause the warning logic to
trigger on new nodes.

Just like changing the version to 8 without doing anything else would be. But it doesn't affect functionality.
The new nodes would only get warnings if the miners, apart from adopting this change were adopting or implementing other change to vote on concrete bits (if they only set nVersion to Consensus::VERSIONBITS_TOP_BITS), no warnings should be raised (unless an unknown SF has been deployed using versionbits, in which case is good that they get the warnings).